src/ngraph/ops/deconvolution.cc

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+#include <vector>
+
+#include "ops/deconvolution.h"
+#include "../../operator/nn/deconvolution-inl.h"
+
+#include "ngraph_emitter.h"
+#include "ngraph_emitter_utils.h"
+#include "ngraph_utils.h"
+
+namespace ngraph_bridge {
+
+NgraphNodePtr create_deconvolution(const NgraphNodePtr& data,
+                                   const NgraphNodePtr& filter,
+                                   const ngraph::Shape& out_shape,
+                                   const nnvm::NodePtr& orig_node) {
+  const auto& param =
+      nnvm::get<mxnet::op::DeconvolutionParam>(orig_node->attrs.parsed);
+  const auto data_shape = data->get_shape();
+  const auto filter_shape = filter->get_shape();
+
+  auto n = data_shape.size() - 2;
```

Comment:
How sure are you that `data_shape.size()` is always >= 2?

If it's ever < 2 the results could be hilarious!

Reply:
Yes, deconvolution only works on 3,4, and 5D tensors.